### DISCUSSION TITLE
Combining saved parameter overrides (toml/yaml file) with --parameter-overrides on the CLI

### DISCUSSION THREAD

**Discussion by mbklein** [aws/aws-sam-cli#6377] [Questions]
I have a project that has both persistent parameters and per-deploy parameters (e.g., one of the lambda functions has to have the current commit hash as an environment variable for error reporting reasons). However, I've noticed that when I use the `--parameter-overrides` command line argument, it completely ignores the `parameter_overrides` section of my `samconfig.yaml` file. I would have expected `sam deploy` to merge parameter values in “immediacy” order, such as:

1. Command line `--parameter-overrides` have highest precedence
2. `samconfig.{toml,yaml}` `parameter_overrides` have next highest precedence
3. Default values have lowest precedence

Is there a way I can achieve what I'm trying to do without either adding ephemeral values to my config file or putting all overrides on the command line?

**Comment by sigJoe** [comment]
Included this in a PR I just opened: #7876

**Reply by vicheey** [reply]
Update: The fixed fro this issue is not longer in https://github.com/aws/aws-sam-cli/pull/7876 since the fix that change default behavior is considered breaking change and need to be handled more carefully. We need a separate fix for this.
